cfme/containers/provider/__init__.py

```diff
@@ -355,6 +361,45 @@ def step(self):
         sel.click(InfoBlock('Overview', 'Topology'))
 
 
+class AdHocMetricsView(BaseLoggedInPage):
+    filter_dropdown = SelectorDropdown('uib-tooltip', 'Filter by')
+    apply_btn = form_buttons.apply_filters
+    selected_filter = None
+
+    @property
+    def is_displayed(self):
+        return False
+
+    def wait_for_filter_option_to_load(self):
+        wait_for(lambda: bool(self.filter_dropdown.items), delay=5, num_sec=60)
+
+    def wait_for_results_to_load(self):
+        wait_for(lambda: bool(int(Text(self, 'h5.ng-binding').text.split()[0])),
```

Comment:
Correct. Unless there is a very specific case that needs to generate locators on the fly, the instantiation of the widgets anywhere else than by widgetastic itself is strictly prohibited.

And 99% of the cases that need to generate locators can do it using the parametrized views.